src/MongoDB.Driver/Linq/Translators/PredicateTranslator.cs

```diff
@@ -1653,8 +1676,8 @@ protected internal override Expression VisitDocument(DocumentExpression node)
 
             protected internal override Expression VisitPipeline(PipelineExpression node)
             {
-                return node;
+                return _useEmptyFieldNameInFieldExpression ? base.VisitPipeline(node) : node;
```

Comment:
This doesn't make sense to me... why would we sometimes call the base method and sometimes not?

I think what this might really mean is that if _useEmptyFieldNameInFieldExpression is true we shouldn't have used this class in the first place.

Thougths @craiggwilson?